### ISSUE TITLE
Feature Requests for 1.0

### ISSUE THREAD

**Issue by Josef-Haupt** [kahst/BirdNET-Analyzer#167]

**Comment by kahst** [comment]
Export re-trained model as Raven model / single model file

**Comment by kahst** [comment]
Support negative training samples for re-training.

- folder names start with "-"
- compute loss for negative samples (inverse loss)
- mask loss for all other classes
- combine positive and negative loss as weighted sum

**Comment by kahst** [comment]
Deal with training samples that are longer than 3-seconds

- split into consecutive 3-second chunks
- allow overlap for splitting

**Comment by kahst** [comment]
Installer for Windows and MacOS

**Comment by kahst** [comment]
Provide meaningful metric for re-training

- current metric implementation cannot deal with "all zero" label vectors, which we use for background classes
- metric also needs to support negative samples once implemented
- maybe move from metric to loss only

**Comment by kahst** [comment]
Signed Windows and MacOS build

See #160

**Comment by kahst** [comment]
Display model version

See #146

**Comment by kahst** [comment]
Adjust max frequency in Raven selection table to sampling rate of source recording

See #140

**Comment by kahst** [comment]
Support multi-file selection review in Raven by adding Begin.File to selection table

See #58

**Comment by kahst** [comment]
Add species.py to GUI

**Comment by kahst** [comment]
What license should a re-trained model have? Can we provide a permissive, commercial-use license for re-trained models?

**Comment by kahst** [comment]
Somehow, results in single file mode vs. multiple file mode differ when using a custom classifier

**Comment by kahst** [comment]
In train.py allow to use Raven selection tables to crop training samples based on annotations from table based on segments.py code.

- train data does not have to be sorted into folders
- folder with train data should contain audio files and one Ravel selection table (or other format we support) for each file
- we extract training samples based on annotations in file
  - we'll pad samples that are too short and split samples that are too long
- segmets.py should contain the code necessary

**Comment by kahst** [comment]
Can we get rid of these warnings when saving a model?

`WARNING:absl:Importing a function (__inference_BLOCK_3-1_ACT_1_layer_call_and_return_conditional_losses_21699) with ops with unsaved custom gradients. Will likely fail if a gradient is requested.`

**Comment by kahst** [comment]
Let people use a separate test dataset instead of splitting from the training data when re-training

**Comment by joshctaylor** [comment]
Is it possible to get a description of how the species filter works, what are the datasources?

**Comment by Mattk70** [comment]
> Is it possible to get a description of how the species filter works, what are the datasources?

Like this: https://github.com/kahst/BirdNET-Analyzer/discussions/234 ?

**Comment by conormcmahon** [comment]
This #234 is really interesting. 
One useful extension would be a simple tool to query that map and let us know whether a chosen point is represented with actual eBird data (and maybe how much by season?) or only with eBird filters. 
E.g. a script which takes a gps point and gives some feedback on how trustworthy the automated species might be.

**Comment by Mattk70** [comment]
Allow custom classifier export to Tensorflow Saved model format.

This would make it far easier to use the custom classifier in TFJS.
